<a href="https://colab.research.google.com/github/SolidusStrange/Matematica-Aplicada/blob/main/011V_G5_Matrices.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import numpy as np

# Problema 1

Las matrices $A$ y $B$ representan las ventas en dos años, en millones de dólares, de los tres productos de una empresa en las cuatro regiones en que está disponible. Es decir, $a_{ij}$ y $b_{ij}$ representan el monto de las ventas del producto $i$ en la región $j$ el primer y segundo año, respectivamente:

<br>

\begin{split}
A = \begin{pmatrix}
 2,6 &  4,8 &  1,8 & 0,9 \\
 3,2 &  4,4 &  2,5 & 2,8 \\
 2,4 &  3,6 &  3,8 & 2,5 \\
\end{pmatrix} \text{ } B= \begin{pmatrix}
 3,6 &  2,5 &  3,0 & 2,5 \\
 4,5 &  5,0 &  3,5 & 3,8 \\
 2,9 &  3,0 &  4,6 & 4,0 \\
\end{pmatrix}
\end{split}

<br>

1. Indique el orden de cada matriz.
2. Calcule la matriz $S = A + B$.
3. Calcule la matriz $D = B - A$.
4. Indique el valor e interprete los elementos $s_{24}$ y $d_{24}$.
5. Para el tercer año de funcionamiento, los tres productos ahora están disponibles en seis regiones. Durante este año, las ventas de esta empresa estarán registradas en una matriz $E$. ¿Es posible determinar la matriz $T = A + B + E$? Justifique su respuesta.

1)

Matriz A 3 x 4

Matriz B 3 x 4

In [ ]:
#pregunta 2
A=np.array([[2.6,4.8,1.8,0.9],
            [3.2,4.4,2.5,2.8],
            [2.4,3.6,3.8,2.5]])
B=np.array([[3.6,2.5,3.0,2.5],
            [4.5,5.0,3.5,3.8],
            [2.9,3.0,4.6,4.0]])

S=A+B
print(f"2) La matriz S es:\n{S}")

#pregunta 3
D=B-A
print(f"\n3) La matriz D es:\n{D}")

#Pregunta 4
print("\n4) s24= 6.6, Indica que la ganancia total de los dos años del producto 2 en la región 4 es de 6.6 millones de dólares")
print("4) d24=1, indica que el año 2, la ganancia del producto 2 en la región 4 fue de 1 millón de dólares más que el año anterior")
print("\n5) No se puede, debido que para poder sumar deben tener todas el mismo orden")

2) La matriz S es:
[[6.2 7.3 4.8 3.4]
 [7.7 9.4 6.  6.6]
 [5.3 6.6 8.4 6.5]]

3) La matriz D es:
[[ 1.  -2.3  1.2  1.6]
 [ 1.3  0.6  1.   1. ]
 [ 0.5 -0.6  0.8  1.5]]

4) s24= 6.6, Indica que la ganancia total de los dos años del producto 2 en la región 4 es de 6.6 millones de dólares
4) d24=1, indica que el año 2, la ganancia del producto 2 en la región 4 fue de 1 millón de dólares más que el año anterior

5) No se puede, debido que para poder sumar deben tener todas el mismo orden


# Problema 2

Una fábrica con sucursales en Japón y China, produce tres tipos de componentes $j$, para cinco modelos de computador $i$. Los costos de producción mensual, en miles de dólares, de ambas sucursales se representan en las matrices $J$ y $C$, respectivamente:

<br>

\begin{split}
J = \begin{pmatrix}
 100 &  200 &  150  \\
 120 &  210 &  170  \\
 112 &  200 &  131  \\
 124 &  310 &  170  \\
 136 &  156 &  140  \\
\end{pmatrix} \text{ } C = \begin{pmatrix}
 110 &  190 &  160  \\
 130 &  220 &  180  \\
 146 &  190 &  141  \\
 143 &  312 &  163  \\
 137 &  116 &  156  \\
\end{pmatrix}
\end{split}

<br>

1. ¿Cuál es el orden de las matrices?
2. Inteprete el elemento $j_{21}$ y $c_{12}$ en el contexto del problema.
3. Utilizando *Python*, calcule el costo total de producción para cada modelo, considerando ambas sucursales.
4. Un trabajador indica que, en la sucursal japonesa, el costo de producción del producto $3$, modelo $4$ de computador, es más alto que en la sucursal china. ¿Tiene razón? Justifique apropiadamente.

In [ ]:
# PREGUNTA 1
# i = filas, j = columnas
# P1 | P2 | P3|
# M1
# M2
# M3

In [ ]:
#pregunta 1
print("1) El orden de las matrices es 5 x 3\n")

#pregunta 2
print("2a) El elemento j21 es 120, Indica que en la planta japonesa, el costo de producción mensual del componente  1  para el modelo de computador  2 , es  120  miles de dólares ")
print("2b) El elemento c12 es 190, Indica que en la planta china, el costo de producción mensual del componente  2  para el modelo de computador  1 , es  190  miles de dólares.\n")

#pregunta 3

J = np.array([[100, 200, 150],
              [120, 210, 170],
              [112, 200, 131],
              [124, 310, 170],
              [136, 156, 140]])

C = np.array([[110, 190, 160],
              [130, 220, 180],
              [146, 190, 141],
              [143, 312, 163],
              [137, 116, 156]])

T = J + C
print(f"3) El costo total de producción de la fabrica es:\n{T}")

#pregunta 4
print("\n4) Es correcto lo que indica el trabajador, ya que el costo de producción mensual del componente  3  para el modelo  4  de computador es  170  mil dólares en la planta japonesa, mientras que en la planta china es  163  mil dólares. Por lo que es  7  mil dólares más caro en la planta japonesa.")

1) El orden de las matrices es 5 x 3

2a) El elemento j21 es 120, Indica que en la planta japonesa, el costo de producción mensual del componente  1  para el modelo de computador  2 , es  120  miles de dólares 
2b) El elemento c12 es 190, Indica que en la planta china, el costo de producción mensual del componente  2  para el modelo de computador  1 , es  190  miles de dólares.

3) El costo total de producción de la fabrica es:
[[210 390 310]
 [250 430 350]
 [258 390 272]
 [267 622 333]
 [273 272 296]]

4) Es correcto lo que indica el trabajador, ya que el costo de producción mensual del componente  3  para el modelo  4  de computador es  170  mil dólares en la planta japonesa, mientras que en la planta china es  163  mil dólares. Por lo que es  7  mil dólares más caro en la planta japonesa.


# Problema 3

La empresa Beta tiene cinco plantas en Chile, y en cada una de ellas fabrica siete productos. El número de unidades del producto $i$ fabricadas en la planta $j$ en una semana está representado en la matriz $C$ en el elemento $c_{ij}$.

<br>

\begin{split}
C = \begin{pmatrix}
 120 &  200 &  155 &  125  &  230 \\
 155 &  130 &  110 &  160  &  180 \\
 140 &  205 &  150 &  125  &  155 \\
 130 &  120 &  230 &  225  &  125 \\
 135 &  70  &  190 &  175  &  95  \\
 110 &  265 &  180 &  230  &  180 \\
 185 &  85  &  120 &  135  &  150 \\
\end{pmatrix}
\end{split}

<br>

La empresa se expande, creando cinco plantas en Ecuador que producen los mismos siete productos. La producción semanal de esta filial estará representada por una matriz $E$.

1. Si se sabe que los volúmenes de producción semanal de la filial en Ecuador son un $20\%$ menor que en Chile, calcule la matriz $E$ utilizando *Python*.
2. Si representamos la producción semanal total de la empresa Beta, considerando ambas filiales, por la matriz $T$, escriba la expresión que calcula cada elemento de la matriz $T$ usando notación matricial.
3. Calcule la matriz $T$ utilizando *Python*.
4. Interprete los elementos $c_{34}$ y $t_{72}$ en el contexto del problema.

In [3]:
    #pl1 pl2 pl3 pl4 i = filas, j = columnas
#p1
#p2
#p3
#p4

In [11]:
# Pregunta 1
C = np.array(
    [[120, 200, 155, 125, 230],
    [155, 130, 110, 160, 180],
    [140, 205, 150, 125, 155],
    [130, 120, 230, 225, 125],
    [135, 70, 190, 175, 95],
    [110, 265, 180, 230, 180],
    [185, 85, 120, 135, 150]])

E = 0.8*C
print(f"1) La matriz E es: \n{E}")


1) La matriz E es: 
[[ 96. 160. 124. 100. 184.]
 [124. 104.  88. 128. 144.]
 [112. 164. 120. 100. 124.]
 [104.  96. 184. 180. 100.]
 [108.  56. 152. 140.  76.]
 [ 88. 212. 144. 184. 144.]
 [148.  68.  96. 108. 120.]]


In [12]:
# Pregunta 2
print("La notación matricial t_ij = c_ij + e_ij")

La notación matricial t_ij = c_ij + e_ij


In [14]:
# Pregunta 3
T = C + E
print(f"La matriz corresponde a: \n{T}")

La matriz corresponde a: 
[[216. 360. 279. 225. 414.]
 [279. 234. 198. 288. 324.]
 [252. 369. 270. 225. 279.]
 [234. 216. 414. 405. 225.]
 [243. 126. 342. 315. 171.]
 [198. 477. 324. 414. 324.]
 [333. 153. 216. 243. 270.]]


In [34]:
print("4) El elemento c_34 es 125, indice que en Chile se fabricaron 125 productos 3 en la fabrica 4")
print("4) El elemento t_72 es 153, indice que en Chile se fabricaron 153 productos 7 en la fabrica 2")

4) El elemento c_34 es 125, indice que en Chile se fabricaron 125 productos 3 en la fabrica 4
4) El elemento t_72 es 153, indice que en Chile se fabricaron 153 productos 7 en la fabrica 2


In [35]:
print(type(T))

<class 'list'>


# Problema 4

Una empresa que fabrica zapatos tiene dos plantas, una en Valparaíso y la otra en Temuco. La empresa produce zapatos de color negro, blanco y café, tanto para niños, damas como caballeros. La capacidad actual de producción mensual (en miles de pares) para Valparaíso y Temuco, están dadas por las matrices $V$ y $T$, respectivamente. Para ambas matrices, según el orden en que se mencionan, las filas corresponden al color del calzado y las columnas al tipo de persona que utilizará el calzado.

<br>

\begin{split}
V = \begin{pmatrix}
 30 &  34 &  20  \\
 45 &  20 &  16  \\
 14 &  26 &  25  \\
\end{pmatrix} \text{ } T = \begin{pmatrix}
 35 &  30 &  26  \\
 52 &  25 &  18  \\
 23 &  24 &  32  \\
\end{pmatrix}
\end{split}

<br>

1. Determine la producción mensual total de ambas plantas, representándola en una matriz $A$. Utilice *Python* para resolver.
2. Interprete el elemento $a_{21}$ en el contexto del problema.
3. Si para el próximo año se estima que la producción en Valparaíso aumenta en un $30\%$, mientras que la de Temuco se reduce en un $10\%$, represente la nueva producción total en la matriz $B$.
4. Interprete el elemento $b_{13}$ en el contexto del problema.

In [3]:
V = np.array([[30,34,20],
              [45,20,16],
              [14,26,25]])

T = np.array([[35,30,26],
              [52,25,18],
              [23,24,32]])

A = V + T
print(f"La producción mensual total de ambas plantas: \n{A}")

La producción mensual total de ambas plantas 
[[65 64 46]
 [97 45 34]
 [37 50 57]]


In [4]:
print("2) La producción total mensual fueron 97000 zapatos blancos para niños")

2) La producción total mensual fueron 97000 zapatos blancos para niños


In [5]:
V2 = V*1.3
T2 = T*0.9
B = V2 + T2
print(f"La nueva total será de: \n{B}")

[[ 70.5  71.2  49.4]
 [105.3  48.5  37. ]
 [ 38.9  55.4  61.3]]


In [ ]:
print("4) La producción el segundo será ")

# Problema 5

Genere una matriz $M$ de orden $8 \times 9$ donde cada término de la matriz está definido por:

$$m_{ij} = 8i + 12j$$

y calcule su transpuesta.

In [7]:
def fun(i, j):
  return 8*(i+1) + 12*(j+1)

M = np.fromfunction(fun,(8,9))
print(f"La matriz M es: \n {M}")


La matriz M es: 
 [[ 20.  32.  44.  56.  68.  80.  92. 104. 116.]
 [ 28.  40.  52.  64.  76.  88. 100. 112. 124.]
 [ 36.  48.  60.  72.  84.  96. 108. 120. 132.]
 [ 44.  56.  68.  80.  92. 104. 116. 128. 140.]
 [ 52.  64.  76.  88. 100. 112. 124. 136. 148.]
 [ 60.  72.  84.  96. 108. 120. 132. 144. 156.]
 [ 68.  80.  92. 104. 116. 128. 140. 152. 164.]
 [ 76.  88. 100. 112. 124. 136. 148. 160. 172.]]


# Problema 6

Genere dos matrices:

*   Una matriz $A$ de orden $5 \times 7$, con $a_{ij} = i \cdot j.$
*   Una matriz $B$ de orden $7 \times 5$, con $b_{ij} = 15i - 2j.$

y calcule $C = A^{T} + \dfrac{1}{2}B$.



# Problema 7

Una red de sensores recoge datos de temperatura ($°C$), humedad (porcentaje) y presión (hPa) en cinco días distintos (de lunes a viernes). Los datos medidos se muestran en la matriz $D$:

<br>

\begin{equation*}
\begin{matrix}
& \begin{matrix}
 & Lu  \ \ &Ma \ \ &Mi \ \ & Ju \ \ &Vi
\end{matrix} \\
&D = \begin{pmatrix}
20    &  28    &  22    &  26     &  24   \\
52    &  57    &  56    &  54     &  58   \\
1.066 &  1.068 &  1.066 &  1.064  &  1.068 \\
\end{pmatrix} & \ \begin{matrix}
Cº\\
\% \\
hPa
\end{matrix}\\
\end{matrix}
\end{equation*}

<br>

Se sabe que por un error de calibración de los sensores, las medidas registradas deberían ser un $5\%$ menores que las registradas.

1. Calcule en una matriz $S$ los valores correctos de temperatura, humedad y presión. Utilice *Python* para calcular.
2. Obtenga la matriz $S^{T}$.
3. Interprete los elementos $s_{24}$ y $s_{31}^{T}$.


# Problema 8

Una empresa tecnológica vende software y hardware en tres productos: $P_{1}$, $P_{2}$ y $P_{3}$, en cuatro zonas de la capital: norte, sur, este y oeste. Las ventas del mes, en millones de pesos, están registradas en las matrices $S$ y $H$, para software y hardware respectivamente.

<br>

\begin{split}
S = \begin{pmatrix}
 260 &  480 &  180  &  90  \\
 320 &  440 &  250  &  280 \\
 240 &  360 &  380  &  250  \\
\end{pmatrix} \text{ } H = \begin{pmatrix}
 360 &  250 &  300  &  250  \\
 450 &  500 &  350  &  380 \\
 290 &  300 &  460  &  400  \\
\end{pmatrix}
\end{split}

<br>

1. Calcule en la matriz $T$, del total de ventas de la empresa tecnológica.
2. Un trabajador indica que este mes, la sucursal de la zona sur recaudó más por la venta hardware que software, ¿es correcta su afirmación?
3. Debido al black friday se espera que el siguiente mes, las ventas de software aumenten un $7\%$ y las de hardware aumenten un $3\%$ respecto al mes actual. Encuentra en total de ventas esperadas para el siguiente mes y almacénalo en la matriz $M$. Utiliza *Python* para realizar los cálculos.
4. Interpreta los elementos $t_{24}$ y $m_{31}$ en el contexto del problema.
5. Calcula la matriz $M^{T}$ e interpreta el elemento $m_{41}^{T}$ en el contexto del problema.
